Reduce size by new url generation and counter reuse

gwer opened the pull request:
Url generation based on @anber_ru's proposal.

Anber commented:
You forgot about `non-secure/index.js` and `url.js`. Also, the comment about low entropy is obsolete :)

ai commented on async/index.browser.js:
```diff
@@ -7,36 +7,25 @@ var crypto = self.crypto || self.msCrypto
     // There is no need in `|| ''` and `* 1.6` hacks in here,
     // because bitmask trim bytes exact to alphabet size.
-    id += url[bytes[size] & 63]
+    id += url[crypto.getRandomValues(new Uint8Array(1)) & 63]
```
This change will kill performance since the single call of `getRandomValues` is much faster than multiple calls.

gwer replied:
![image](https://user-images.githubusercontent.com/6529684/73334212-2f4ad500-427d-11ea-8042-9e1c7ab6c5c8.png)

ai commented on async/index.browser.js:
```diff
@@ -7,36 +7,30 @@ var crypto = self.crypto || self.msCrypto
 // Despite the fact the source code is quite long, its entropy
 // is low and there are lots of duplicates - just what compressors
```
You need to update comment too

ai commented on async/index.browser.js:
```diff
@@ -7,36 +7,30 @@ var crypto = self.crypto || self.msCrypto
-  i += 1, i += 1, i += 1, i += 1, i += 1, i += 1, i += 1, i += 1,
-  i += 1, i += 1, i += 1, i += 1, i += 1, i += 1, i += 1, i += 1,
-  i += 1, i += 1,
+var i = 36
```
Let’s comment on what we generate here

ai commented on async/index.browser.js:
```diff
@@ -7,36 +7,30 @@ var crypto = self.crypto || self.msCrypto
-  i += 1, i += 1
-)
+i = 36
+while (i-- - 10) {
```
We need a comment about this `while` loop

ai commented on async/index.browser.js:
```diff
@@ -7,36 +7,30 @@ var crypto = self.crypto || self.msCrypto
-  i += 1, i += 1,
+var i = 36
+while (i--) {
+  url += i.toString(36)
```
Let’s comment `36` argument of `toString`

gwer replied:
Done

Anber commented on async/index.browser.js:
```diff
@@ -4,39 +4,38 @@
+  // 36 is radix.
+  // Number.prototype.toString(36) returns number in Base36 representation
+  // Base36 is like hex,
+  // but Base36 is represented using the numerals 0–9 and the Latin letters A–Z
```
Latin letters ***a-z***

gwer replied:
Вообще писал про Base36 в целом (где понятия регистра вообще нет), но, пожалуй, тут описание ближе к реализации из JS будет корректнее, да.

ai commented:
Ещё нужно сравнить результаты `test/benchmark` до и после этого PR

gwer commented:
```
$ git checkout master
Switched to branch 'master'
Your branch is up to date with 'origin/master'.

$ node test/benchmark
nanoid                    282,863 ops/sec
nanoid/generate           255,983 ops/sec
uid.sync                  170,236 ops/sec
uuid/v4                   202,241 ops/sec
secure-random-string      189,137 ops/sec
shortid                    30,185 ops/sec

Async:
nanoid/async               31,993 ops/sec
nanoid/async/generate      31,449 ops/sec
secure-random-string       30,437 ops/sec
uid                        29,401 ops/sec

Non-secure:
nanoid/non-secure       1,714,070 ops/sec
rndm                    1,593,879 ops/sec

$ git checkout -
Switched to branch 'f/reduce_size_with_random_in_loop'
Your branch is up to date with 'origin/f/reduce_size_with_random_in_loop'.

$ node test/benchmark
nanoid                    294,016 ops/sec
nanoid/generate           265,646 ops/sec
uid.sync                  201,971 ops/sec
uuid/v4                   219,655 ops/sec
secure-random-string      183,223 ops/sec
shortid                    28,493 ops/sec

Async:
nanoid/async               33,153 ops/sec
nanoid/async/generate      30,710 ops/sec
secure-random-string       28,456 ops/sec
uid                        28,692 ops/sec

Non-secure:
nanoid/non-secure       1,552,124 ops/sec
rndm                    1,683,481 ops/sec
```

ai commented:
Released in 2.1.11

alfaproject commented on async/index.browser.js:
```diff
@@ -4,39 +4,38 @@
   var id = ''
-  var bytes = crypto.getRandomValues(new Uint8Array(size))
+  var bytes = crypto.getRandomValues(new Uint8Array(size || 21))
+  i = size || 21
```
Why not?
```js
  i = size || 21
  var bytes = crypto.getRandomValues(new Uint8Array(i))
```

gwer replied:
For better gzip compression. It makes +1 byte.

alfaproject replied:
Really? o:

gwer replied:
Sorry, I was wrong. +3 bytes :D

alfaproject replied:
Haha
